### ISSUE TITLE
"TypeError: Cannot read property 'type' of undefined" when clicking "Go to about page via redux"

### ISSUE THREAD

**Issue by joncass** [notrab/create-react-app-redux#1]
Hello -

I like the setup of your project a lot and found your [Medium post](https://medium.com/@notrab/getting-started-with-create-react-app-redux-react-router-redux-thunk-d6a19259f71f) very useful, so thanks!

When I clone this repo and start it, I am having the issue from the title:
`"TypeError: Cannot read property 'type' of undefined"`
when I click the "Go to about page via redux".

Based on [this commit on cyberid41's fork](https://github.com/cyberid41/create-react-app-redux/commit/5c7733a61db369d0649019da178c355d2abbf451), they seem to have the issue too.

Some googling led me to [this](https://github.com/reactjs/react-router-redux/issues/493) issue on the react-router-redux page, where it was indicated there might be a problem with the ordering of the middleware, but it looks right to me (thunk first).

Anyway, you obviously don't owe me anything but I figured you might want to know and if you do happen to have a solution I'd love to see it :). Also, I'm very new to open source and github, so apologies if this is an incorrect or inappropriate way to file an issue. Thanks for reading!

**Comment by notrab** [comment]
Hey @joncass 

This is the perfect place for these kind of issues. So don't worry! I'm new to OSS too, so we're all learning 😀

I'll have a look into the issue and get back to you shortly. Sorry about the inconvenience!

**Comment by patientplatypus** [comment]
I as well had the same issue (error page screenshot http://imgur.com/a/OxAxI). I would also like to note that you're code and the code on the tutorial I followed were different (and I assumed that the github was more up to date). FYI. Thanks for posting the tutorials.

**Comment by notrab** [comment]
Hey @patientplatypus

Thanks for the feedback, I'll update the Medium post this weekend and change the way the dispatch is used for changing routes. 

Have a great weekend!

**Comment by notrab** [comment]
Ok so the problem with this is the use of `bindActionCreators`.

When I first made the tutorial I wasn't use this and I dropped it in last minute so it's failing due to calling dispatch on something that is already mapped to a dispatch.

Inside `src/containers/home` you'll want to change `changePage` function to be....

```js
changePage: () => push('/about-us')
```

I'll update the tutorial + repo next.
